src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -156,7 +150,15 @@ private void EnableTimer(float pollingIntervalInSeconds)
                         restoreFlow = true;
                     }
 
-                    _pollingTimer = new Timer(s => ((CounterGroup)s!).OnTimer(null), this, _pollingIntervalInMilliseconds, _pollingIntervalInMilliseconds);
+                    if (_pollingThread == null)
```

Comment:
The suppress flow dance above is only needed for creating the new thread. It is not needed for setting the event.